src/robot/parsing/lexer/blocklexers.py

```diff
@@ -52,14 +52,15 @@ def input(self, statement):
         return lexer
 
     def lexer_for(self, statement):
-        for cls in self.lexer_classes():
-            lexer = cls(self.ctx)
-            if lexer.handles(statement):
+        for cls in self.RESOLVED_LEXER_CLASSES:
```

Comment:
This was the major improvement: not needing to create instances to use `handles` and not creating temporary tuples from the call to `lexer_classes()`.

This meant that the `lexer_classes` and `handles` had to be changed to `classmethods`.